dig_test.go

```diff
@@ -286,6 +286,53 @@ func TestEndToEndSuccess(t *testing.T) {
 		}))
 	})
 
+	t.Run("param wrapper", func(t *testing.T) {
+		c := New()
+		require.NoError(t, c.Provide(func() *bytes.Buffer {
+			return new(bytes.Buffer)
+		}), "provide failed")
+
+		type MyParam struct{ Param }
+
+		type Args struct {
+			MyParam
+
+			Buffer *bytes.Buffer
+		}
+
+		require.NoError(t, c.Invoke(func(args Args) {
+			require.NotNil(t, args.Buffer, "invoke got nil buffer")
+		}))
+	})
+
+	t.Run("param recurse", func(t *testing.T) {
+		type anotherParam struct {
+			Param
+
+			Buffer *bytes.Buffer
+		}
+
+		type someParam struct {
+			Param
+
+			Buffer  *bytes.Buffer
+			Another *anotherParam
+		}
+
+		c := New()
+		require.NoError(t, c.Provide(func() *bytes.Buffer {
+			return new(bytes.Buffer)
+		}), "provide must not fail")
+
+		require.NoError(t, c.Invoke(func(p *someParam) {
+			require.NotNil(t, p, "someParam must not be nil")
+			require.NotNil(t, p.Buffer, "someParam must not be nil")
+			require.NotNil(t, p.Another, "someParam must not be nil")
+			require.NotNil(t, p.Another.Buffer, "someParam must not be nil")
+			require.True(t, p.Buffer == p.Another.Buffer, "buffer must be the same")
+		}), "invoke must not fail")
+	})
```

Comment:
Can we also validate that (1) we're getting the same buffer reference that the constructor returned, and (2) we're only calling the buffer constructor once? Same for test above.